Data node setup: Add history retention flags & fetch info (#459)
vegaprotocol/documentation: docs/node-operators/get-started/setup-datanode.md

:::

### Data node configuration

## Data node retention profiles
When starting a data node, you can choose the data retention configuration for your data node, depending on the use case for the node. The retention policy details can all be fine-tuned manually, as well.

There are 3 retention policy configurations: 
* **Standard (default)**: The node retains data according to the default retention policies, which assume a data node retains some data over time, but not all data
* **Lite**: The node retains enough data to be able to provide the latest state to clients, and produce network history segments. This mode saves enough to provide the current state of accounts, assets, balances, delegations, liquidity provisions, live orders, margin levels, markets, network limits, network parameters, node details, parties, positions 
* **Archive**: The node retains all data

To run a node that doesn't use the standard default retention, use one of the following flags when running the `init` command: 
   
* For a standard node, no flag
* For an archive node, use `--archive`
* For a lite node, use `--lite`

> 

Author: ```suggestion

If you want to tweak the retention policy once the initial configuration has been generated, set it on per-table basis in the data node's `config.toml`. 

For example: 

```toml
[[SQLStore.RetentionPolicies]]
  HypertableOrCaggName = "balances"
  DataRetentionPeriod = "7 days"
```
```

> ## Generate config

Reviewer: You might perhaps want to add that if you want to tweak the retention policy once the initial configuration has been generated, then one can set it on per-table basis in the data-node's config.toml in the sections like
```toml
  [[SQLStore.RetentionPolicies]]
    HypertableOrCaggName = "balances"
    DataRetentionPeriod = "7 days"
```

Author: Nice - I will add that, thank you!

vega start --home="YOUR_VEGA_HOME_PATH" --tendermint-home="YOUR_TENDERMINT_HOME_PATH"
```

## Fetch network history
After starting a data node, you can load in a segment of network history, if you want your node to have more data than provided by the current height. This is particularly useful if you're running an archive node.

To see how much network history your data node has, run the following command:

```shell
vega datanode network-history show --home="YOUR_DATA_NODE_HOME_PATH"
```

> To fetch a network history segment, run the command below. Use the ID of the segment you want (for example, the oldest) and load all data from a block height. `500` is used as an example.

Reviewer 2: ```suggestion
To fetch a network history segment, run the command below. Use the ID of the segment you want (for example, the oldest) followed by the number of blocks prior to the segments height that you want fetch. `2000` is used in the following example.  This will result in all blocks from height 3000 to 5000 being retrieved.
```

```shell

> vega datanode fetch <segment-id-of-segment-at-height-500> 500

Reviewer 2: ```suggestion
vega datanode fetch <segment-id-of-segment-at-height-5000> 2000
```
